src/webview/connections.ts

```diff
@@ -0,0 +1,138 @@
+// @ts-nocheck
+// This script will be run within the webview itself
+// It cannot access the main VS Code APIs directly.
+(function () {
+  const vscode = acquireVsCodeApi();
+
+  document.querySelector('.add-connection-button')?.addEventListener('click', () => {
+    addConnection();
+  });
+
+  // Handle messages sent from the extension to the webview
+  window.addEventListener('message', event => {
+    const message = event.data; // The json data that the extension sent
+    switch (message.type) {
+      case 'cql.connections.clearConnections': {
+        values = [];
+        updateConnectionList(values);
+        break;
+      }
+      case 'Connections.refreshConnections': {
+        refreshConnections();
+        break;
+      }
+      case 'Connections.createConnectionsView': {
+        createConnectionsView(message.connections, message.currentConnection);
+        break;
+      }
+    }
+  });
+
+  function updateConnectionList(connections, currentConnection) {
+    console.log('Type of connection is ' + typeof connections);
+    const connectionsList = document.querySelector('.connections-list');
+    connectionsList.textContent = '';
+    if (Object.keys(connections).length === 0) {
+      connectionsList.textContent = 'You have no saved connections. Add one below.';
+    }
+    if (connectionsList) {
+      if (connections !== undefined) {
```

Comment:
These two if-statements can be combined and replaced with an early return:

```javascript
if (!connectionsList || connections === undefined) {
  return;
}
```

This reduces the level of nesting, making for more readable code. There's a nice brief video on this rationale [here](https://www.youtube.com/watch?v=CFRhGnuXG-4).